src/Sparrow/Json/BlittableWriter.cs

```diff
@@ -136,8 +140,10 @@ public WriteToken WriteObjectMetadata(List<PropertyTag> properties, long firstWr
             _position += WriteVariableSizeInt(properties.Count);
 
             // Write object metadata
-            foreach (var sortedProperty in properties)
+            for (int i = 0; i < properties.Count; i++)
```

Comment:
Yes and No. They are fast because they do not allocate, but they are not fast enough for very hot paths because the MoveNext and other calls are not inlined.